challengeutils/wiki.py

```diff
@@ -0,0 +1,208 @@
+        Wiki Configuration
+        [
+            {
+                "id": "111",
+                "title": "title",
+                "parentId": "33333",
+                "markdown_path": "home.md"
+            },
+            {...}
+        ]
+    """
+    config_path = os.path.join(workdir, "wiki_config.json")
+    if not os.path.exists(config_path):
+        raise ValueError("wiki_config.json must exist at specified workdir")
+    with open(config_path, "r") as config:
+        wiki_config = json.load(config)
+    return wiki_config
+
+
+def validate_config(workdir: str) -> typing.List[dict]:
+    """Validates wiki configuration
+
+    Args:
+        workdir: Workfing directory with markdown and wiki_config.json
+
+    Returns:
+        Wiki Configuration
+        [
+            {
+                "id": "111",
+                "title": "title",
+                "parentId": "33333",
+                "markdown_path": "home.md"
+            },
+            {...}
+        ]
+
+    Raises:
+        ValueError:
+            `markdown_path` is specified but cannot be located.
+            There are duplicated wiki ids.
+            `id` is not specified and (`markdown_path`/`parentId`/`title`
+            is missing or `parentId` not one of the `id`s.)
+            `id` cannot be the same as `parentId`.
+
+    """
+    wiki_config = read_wiki_config(workdir)
+    ids = [str(wiki_header['id']) for wiki_header in wiki_config
+           if wiki_header.get('id') is not None]
+    if len(set(ids)) != len(ids):
+        raise ValueError("Must have unique ids.")
+
+    # Must have one home page where `parentId` is blank
+    home_page = [wiki_header.get('parentId') is None
+                 for wiki_header in wiki_config]
+    if sum(home_page) != 1:
+        raise ValueError("Must only have one config "
+                         "where `parentId` is blank")
+    for wiki_header in wiki_config:
+        # Configuration must not be empty
+        markdown_path = wiki_header.get('markdown_path')
+        wikiid = wiki_header.get('id')
+        parentid = wiki_header.get('parentId')
+        title = wiki_header.get('title')
+        # Title must be specified
+        if title is None:
+            raise ValueError("Must have title")
+        title = str(title).strip()
+
+        # Make id and parentid strings and strip white spaces if values
+        # are specified
+        if wikiid is not None:
+            wikiid = str(wikiid).strip()
+            wiki_header['id'] = wikiid
+        if parentid is not None:
+            parentid = str(parentid).strip()
+            wiki_header['parentId'] = parentid
+
+        # id and parentid must not be empty strings
+        if wikiid == '' or parentid == '' or title == '':
+            raise ValueError('`id`, `parentId`, and `title` must not be '
+                             'empty strings if specified')
+        # Markdown file must exist if specified
+        if markdown_path is not None and not os.path.exists(
```

Comment:
Just confirming again whether it is okay for `markdown_path` to be an empty string? I'm assuming the wiki page will just be an empty page in this case.

Reply:
if markdown is an empty string is will fail since an empty string doesn't exist as a file path.
```python
import os
markdown_path = ""
markdown_path is not None and not os.path.exists(markdown_path)
True


```

Reply:
Good catch! I was originally going to say that isn't an issue, but I realized i was doing 

```
os.path.exists(workdir...
```

I changed to `os.path.isfile` instead, so an empty string should fail.

Reply:
Ah, good solution!